src/Commands/Site/DeleteCommand.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Site;
+
+/**
+ * Class DeleteCommand
+ * @package Pantheon\Terminus\Commands\Site
+ */
+class DeleteCommand extends SiteCommand
+{
+    /**
+     * Deletes a site from Pantheon
+     *
+     * @authorized
+     *
+     * @command site:delete
+     *
+     * @option string $site_name Name of the site to delete
+     * @usage terminus site:delete <site_name>
```

Comment:
Minor: we should probably bless a format for `@usage` examples, I see `<site_name>`, `mysite` and also `awesome-site`. I personally like this `<site_name>` the best.